packages/multichain-account-service/src/providers/BaseBip44AccountProvider.ts

```diff
@@ -116,13 +137,16 @@ export abstract class BaseBip44AccountProvider<
    * @throws If the account is not found.
    */
   getAccount(id: Account['id']): Account {
-    const found = this.getAccounts().find((account) => account.id === id);
+    const hasAccount = this.accounts.has(id);
 
-    if (!found) {
+    if (!hasAccount) {
       throw new Error(`Unable to find account: ${id}`);
     }
 
-    return found;
+    return this.messenger.call(
```

Comment:
Feel free to update the wording, but adding a comment is worth here, IMO:

```suggestion
    // We need to up-cast here, since `InternalAccount` are not always `Bip44Account`s, though
    // we know this account is BIP-44 compatible, so it's safe to cast.
    return this.messenger.call(
```